test/org/zaproxy/zap/authentication/UsernamePasswordAuthenticationCredentialsTest.java

```diff
@@ -0,0 +1,115 @@
+package org.zaproxy.zap.authentication;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseSet;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+
+public class UsernamePasswordAuthenticationCredentialsTest {
+
+    private UsernamePasswordAuthenticationCredentials usernamePasswordAuthenticationCredentials;
+    private ApiResponseSet apiResponseSet;
+    private String username = "username";
+    private String password = "password";
+
+    @Before
+    public void setUp() {
+        this.usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+    }
+
+    @Test
+    public void shouldNotUsernameBeEmptyOrNull() throws Exception {
```

Comment:
Probably better to rename the test, as is it might give the impression that the `UsernamePasswordAuthenticationCredentials` is created always with a non null/empty username (which might not be the case).